src/Catalog/AggregateCursor.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    /// <summary>
+    /// A <see cref="ReadCursor"/> that returns the least value for <see cref="Load(CancellationToken)"/> from a set of <see cref="ReadCursor"/>s.
+    /// </summary>
+    public class AggregateCursor : ReadCursor
+    {
+        public AggregateCursor(IEnumerable<ReadCursor> innerCursors)
+        {
```

Comment:
`innerCursors.Count()` returns `0` if `innerCursors` is null, thus it is checked